lectures/5-commitments.tex

```diff
@@ -0,0 +1,294 @@
+\documentclass[../lecture-notes.tex]{subfiles}
+
+\begin{document}
+
+\subsection{Commitments}
+
+\begin{definition}
+    A cryptographic commitment scheme allows one party to commit to a chosen statement (such as a value, vector, or polynomial) without revealing the statement itself. The commitment can be revealed in full or in part at a later time, ensuring the integrity and secrecy of the original statement until the moment of disclosure.
+\end{definition}
+
+Before delving into the details, here is the intuition of cryptographic commitments.
+
+Imagine putting a letter with some message into a box and locking it with your key. 
+You then give that box to your friend, who cannot open it without the key.
+In this scenario, you have made a commitment to the message inside the box. 
+You cannot change the content of the letter, as it is in your friend's possession. 
+At the same time, your friend cannot access the letter since they do not have the key to unlock the box.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.5\linewidth, clip]{images/lecture_5/CommitmentExample.png}
+
+    \caption{Commitment scheme}
+\end{figure}
+
+\begin{definition}[Commitment Scheme]
+    
+    Commitment Scheme $\Pi_{\text{commitment}}$ is a tuple of three algorithms: $\Pi_{\text{commitment}} = (\mathsf{Setup}, \mathsf{Commit}, \mathsf{Verify})$.
+
+    \begin{enumerate}
+
+        \item $\mathsf{Setup}$ ($1^{\lambda}$): returns public parameter $\mathsf{pp}$ for both comitter and verifier;
```

Comment:
Better to write $\mathsf{Setup}(1^{\lambda})$ is a single $$ block, but let it be as it is